store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java

```diff
@@ -258,12 +269,38 @@ public void maybeExpandInSyncStateSet(final String slaveAddress, final long slav
             final EpochEntry currentLeaderEpoch = this.epochCache.lastEntry();
             if (slaveMaxOffset >= currentLeaderEpoch.getStartOffset()) {
                 currentSyncStateSet.add(slaveAddress);
+                markSynchronizingSyncStateSet(currentSyncStateSet);
                 // Notify the upper layer that syncStateSet changed.
                 notifySyncStateSetChanged(currentSyncStateSet);
             }
         }
     }
 
+    private void markSynchronizingSyncStateSet(final Set<String> newSyncStateSet) {
+        this.writeLock.lock();
+        try {
+            this.isSynchronizingSyncStateSet = true;
+            this.remoteSyncStateSet.clear();
+            this.remoteSyncStateSet.addAll(newSyncStateSet);
+        } finally {
+            this.writeLock.unlock();
+        }
+    }
+
+    private void markSynchronizingSyncStateSetDone() {
+        this.writeLock.lock();
+        try {
+            this.isSynchronizingSyncStateSet = false;
+            this.remoteSyncStateSet.clear();
+        } finally {
+            this.writeLock.unlock();
+        }
+    }
```

Comment:
lock not must. this method is invoked by setSyncStateSet method that has write lock
```
    public void setSyncStateSet(final Set<String> syncStateSet) {
        this.writeLock.lock();
        try {
            markSynchronizingSyncStateSetDone();
            this.syncStateSet.clear();
            this.syncStateSet.addAll(syncStateSet);
            this.confirmOffset = computeConfirmOffset();
        } finally {
            this.writeLock.unlock();
        }
    }
```